packages/components/src/Gallery/Gallery.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from "react";
+import styles from "./Gallery.css";
+import { File, GalleryProps } from "./GalleryTypes";
+import { LightBox } from "../LightBox";
+import { FormatFile } from "../FormatFile";
+import { Button } from "../Button";
+
+export function Gallery({ files, size = "base", max, onDelete }: GalleryProps) {
+  const { images, filesToImageIndex } = generateImagesArray(files);
+
+  const [lightboxOpen, setLightboxOpen] = useState(false);
+  const [lightboxIndex, setLightboxIndex] = useState(0);
+  const [displayPastMax, setDisplayPastMax] = useState(false);
+
+  return (
+    <>
+      <div className={size === "large" ? styles.galleryLarge : styles.gallery}>
+        {files.map((file, index) => {
+          if (max && !displayPastMax && index + 1 > max) {
+            return <></>;
+          }
```

Comment:
This is gonna thrown an error on react since it doesn't have a key. TBH, not a fan of this since we're doing every bit of logic in JSX and that's a code smell.

What I would suggest is to add a const that determines whether to map all the files or only the first `n` files.

_Par exemple_
```tsx
const visibleFiles = displayPastMax ? files : files.slice(0, max);
```

This way you won't have to do the `if (max && !displayPastMax && index + 1 > max) {` logic since it won't render what's not there and fixes the key error that react throws.

Tho you would have to adjust the default value for `displayPastMax`.